src/Cygni.Snake.Client/Map.cs

```diff
@@ -63,28 +87,49 @@ public IEnumerable<MapCoordinate> SnakeParts
             }
         }
 
+        public DirectionalResult GetResultOfMyDirection(Direction dir)
+        {
+            var target = MySnake.HeadPosition.GetDestination(dir);
+
+            switch (this[target])
+            {
+                case TileType.Food:
+                    return DirectionalResult.Food;
+                case TileType.Empty:
+                case TileType.OpponentHeadNeighbor:
```

Comment:
Worth considering a "Danger" DirectionalResult to signify when you are close to an opponent's head.